### PR TITLE
let, var and const challenge

### PR DESCRIPTION
**Pull request by rjimmy2512**

### CODE REVIEW

**Comment by rjimmy2512** [comment]
Solved

File: curriculum/class-00/lab-a/starter-code/js/app.js [hunk 1]
```diff
@@ -1,136 +1,140 @@
...
+      viewed.push(rando);
+    }
+  }
+  // TODO: In a sentence or two, explain why the previous line of code threw an error when we changed the constiable declaration from `let to `let`.
+  // PUT YOUR RESPONSE IN THIS COMMENT
+  //EXPLANATION: This line is giving an error because rando is defined to be block-scoped inside the function displayPics as it has been defined by using 'let', but here it is called (console-logged) outside the function like a global variable.
+
+  console.log(viewed);
+
+  for (var i = 0; i < 3; i++){
```
Comment (Tw1stedRain) [inline curriculum/class-00/lab-a/starter-code/js/app.js]: looks like you missed a `var`, make sure you are using `let` or `const` for everything form here on out

File: curriculum/class-00/lab-a/starter-code/js/app.js [hunk 2]
```diff
@@ -1,136 +1,140 @@
...
+function displayPics(){
+  while(viewed.length < 6){
+    let rando = makeRandom();
+    while(!viewed.includes(rando)){
+      viewed.push(rando);
+    }
+  }
+  // TODO: In a sentence or two, explain why the previous line of code threw an error when we changed the constiable declaration from `let to `let`.
+  // PUT YOUR RESPONSE IN THIS COMMENT
+  //EXPLANATION: This line is giving an error because rando is defined to be block-scoped inside the function displayPics as it has been defined by using 'let', but here it is called (console-logged) outside the function like a global variable.
```
Comment (Tw1stedRain) [inline curriculum/class-00/lab-a/starter-code/js/app.js]: this is a perfect explanation for why this happened

File: curriculum/class-00/lab-a/starter-code/js/app.js [hunk 3]
```diff
@@ -1,136 +1,140 @@
...
+    container.removeEventListener('click', handleClick);
+    container.style.display = 'none';
+    showList();
+    makeChart();
+  }
+  for(let i = 0; i < names.length; i++){
+    if(event.target.id === allProducts[i].name) {
+      allProducts[i].votes += 1;
+      // console.log(event.target.id + ' has ' + allProducts[i].votes + ' votes in ' + allProducts[i].views + ' views');
+      console.log(event.target.id + ' has ' + allProducts[i].votes + ' votes ');
```
Comment (Tw1stedRain) [inline curriculum/class-00/lab-a/starter-code/js/app.js]: you did not change your concatenations into template literals

File: curriculum/class-00/lab-a/starter-code/js/app.js [hunk 4]
```diff
@@ -1,136 +1,140 @@
...
+  }
+  localStorage.busmall = JSON.stringify(allProducts);
+  localStorage.busmallProducts = JSON.stringify(allProducts);
+  displayPics();
+}
+
+function showList() {
+  for(let i = 0; i < allProducts.length; i++) {
+    let liEl = document.createElement('li');
+    liEl.textContent = allProducts[i].name + ' has ' + allProducts[i].votes + ' votes and ' + allProducts[i].name + ' has ' + allProducts[i].views + ' views';
```
Comment (Tw1stedRain) [inline curriculum/class-00/lab-a/starter-code/js/app.js]: you did not change your concatenations into template literals

**Pull request closed without merging**